syntax = "proto3";

package ecommerce;

service OrderManagement {
    rpc addOrder(Order) returns (StringMessage);
    rpc getOrder(StringMessage) returns (Order);
    rpc searchOrders(StringMessage) returns (stream Order);
    rpc processOrders(stream Order) returns (stream CombinedShipment);
}

message Order {
    string id = 1;
    repeated string items = 2;
    string description = 3;
    float price = 4;
    string destination = 5;
}

message CombinedShipment {
    string id = 1;
    string status = 2;
    repeated Order orders = 3;
}

message StringMessage {
    string value = 1;
}
